File: backend_fastapi/schemas/pet.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class CreatePetRequest(BaseModel):
    name: str
    appearance: str
    personality: str
    color: str


class PetResponse(BaseModel):
    id: int
    name: str
    appearance: str
    personality: str
    color: str
    created_at: Optional[datetime] = None
    last_interacted_at: Optional[datetime] = None

    class Config:
        from_attributes = True

File: server-deploy/backend/src/main/java/com/example/glimmerseed/entity/Pet.java
package com.example.glimmerseed.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "pets")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Pet {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(nullable = false)
    private String name;
    
    @Column(nullable = false)
    private String appearance;
    
    @Column(nullable = false)
    private String personality;
    
    @Column(nullable = false)
    private String color;
    
    @Column(name = "user_id")
    private Long userId;
    
    @Column(name = "created_at")
    private LocalDateTime createdAt;
    
    @Column(name = "last_interacted_at")
    private LocalDateTime lastInteractedAt;
    
    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        lastInteractedAt = LocalDateTime.now();
    }
}